flink/src/main/java/org/apache/iceberg/flink/actions/MigrateAction.java

```diff
@@ -0,0 +1,374 @@
+      } catch (Exception e) {
+        throw new RuntimeException(e);
+      }
+    }
+
+    AppendFiles append = icebergTable.newAppend();
+    files.forEach(append::appendFile);
+    append.commit();
+    return files.size();
+  }
+
+  private Namespace toNamespace(Namespace namespace) {
+    if (namespace.isEmpty()) {
+      return Namespace.of(icebergDbName);
+    } else {
+      String[] namespaces = new String[baseNamespace.levels().length + 1];
+      System.arraycopy(baseNamespace.levels(), 0, namespaces, 0, baseNamespace.levels().length);
+      namespaces[baseNamespace.levels().length] = icebergDbName;
+      return Namespace.of(namespaces);
+    }
+  }
+
+  private List<DataFile> importPartitions(PartitionSpec spec, ObjectPath tableSource,
+                                          String nameMapping, FileFormat fileFormat,
+                                          MetricsConfig metricsConfig, String hiveTableName)
+      throws Exception {
+    List<CatalogPartitionSpec> partitionSpecs = flinkHiveCatalog.listPartitions(tableSource);
+    List<PartitionAndLocation> inputs = Lists.newArrayList();
+    for (CatalogPartitionSpec partitionSpec : partitionSpecs) {
+      Partition partition =
+          flinkHiveCatalog.getHivePartition(flinkHiveCatalog.getHiveTable(tableSource), partitionSpec);
+      inputs.add(
+          new PartitionAndLocation(partition.getSd().getLocation(), Maps.newHashMap(partitionSpec.getPartitionSpec())));
+    }
+
+    int size = partitionSpecs.size();
+    int parallelism = Math.min(size, maxParallelism);
+
+    DataStream<PartitionAndLocation> dataStream = env.fromCollection(inputs);
+    MigrateMap migrateMap = new MigrateMap(spec, nameMapping, fileFormat, metricsConfig);
+    DataStream<List<DataFile>> ds = dataStream.map(migrateMap).setParallelism(parallelism);
+    return Lists.newArrayList(ds.executeAndCollect("migrate table :" + hiveTableName)).stream()
+        .flatMap(Collection::stream).collect(Collectors.toList());
+  }
+
+  private static class MigrateMap implements MapFunction<PartitionAndLocation, List<DataFile>> {
+    private final PartitionSpec spec;
+    private final String nameMappingString;
+    private final FileFormat fileFormat;
+    private final MetricsConfig metricsConfig;
+
+    MigrateMap(PartitionSpec spec, String nameMapping, FileFormat fileFormat, MetricsConfig metricsConfig) {
+      this.spec = spec;
+      this.nameMappingString = nameMapping;
+      this.fileFormat = fileFormat;
+      this.metricsConfig = metricsConfig;
+    }
+
+    @Override
+    public List<DataFile> map(PartitionAndLocation map) {
+      Map<String, String> partitions = map.getMap();
+      String location = map.getLocation();
+      Configuration conf = HadoopUtils.getHadoopConfiguration(GlobalConfiguration.loadConfiguration());
+      NameMapping nameMapping = nameMappingString != null ? NameMappingParser.fromJson(nameMappingString) : null;
+      List<DataFile> files;
+      switch (fileFormat) {
```

Comment:
could a table have mixed file formats?

Reply:
I lookup the [hive doc](https://cwiki.apache.org/confluence/display/Hive/LanguageManual+DDL#LanguageManualDDL-CreateTableCreate/Drop/TruncateTable), when we  create  a hive table , specify the table format by `STORED AS file_format`, and the table format is table level properties. I think we cannot create different formats for the same table.

Reply:
One can change the InputFormat/OutputFormat, but that will result in a "dead" table. So it is ok to expect only a single file format.